### ISSUE TITLE
Implement softmax

### ISSUE THREAD

**Issue by ypapadop-amd** [ypapadop-amd/ggml#124]
This is needed for cross entropy loss tests.
